src/mlpack/methods/random_forest/extra_trees_main.cpp

```diff
@@ -0,0 +1,263 @@
+        { return x >= 0; },
+        true,
+        "maximum depth must not be negative");
+    RequireParamValue<int>(
+        "subspace_dim", [](int x)
+        { return x >= 0; },
+        true,
+        "subspace dimensionality must be nonnegative");
+    RequireParamValue<double>(
+        "minimum_gain_split",
+        [](double x)
+        { return x >= 0.0; },
+        true,
+        "minimum gain for splitting must be nonnegative");
+    ReportIgnoredParam({{"training", false}}, "num_trees");
+    ReportIgnoredParam({{"training", false}}, "minimum_leaf_size");
+    ExtraTreesModel *etModel;
+    if (IO::HasParam("input_model"))
+        etModel = IO::GetParam < ExtraTreesModel*> ("input_model");
+    else
+        etModel = new ExtraTreesModel();
+    if (IO ::HasParam("training"))
+    {
+        Timer::Start("et_training");
+        // Training the model on the given input data.
+        mat data = std::move(IO::GetParam<mat>("training"));
+        Row<size_t> labels = std::move(IO::GetParam<Row<size_t>>("labels"));
+        // Make sure the subspace dimensionality is valid.
+        RequireParamValue<int>(
+            "subspace_dim",
+            [data](int x)
+            { return (size_t)x <= data.n_rows; },
+            true, "subspace "
+                  "dimensionality must not be greater than data dimensionality");
+        const size_t numTrees = (size_t)IO::GetParam<int>("num_trees");
+        const size_t minimumLeafSize =
+            (size_t)IO::GetParam<int>("minimum_leaf_size");
+        const size_t maxDepth = (size_t)IO::GetParam<int>("maximum_depth");
+        const double minimumGainSplit = IO::GetParam<double>("minimum_gain_split");
+        const size_t randomDims = (IO::GetParam<int>("subspace_dim") == 0) ? 
+        (size_t)std::sqrt(data.n_rows) : (size_t)IO::GetParam<int>("subspace_dim");
+        MultipleRandomDimensionSelect mrds(randomDims);
+
+        Log::Info << "Training extra trees with " << numTrees << " trees..."
+                  << endl;
+        const size_t numClasses = max(labels) + 1;
+        // Training the model.
+        etModel->et.Train(data, labels, numClasses, numTrees, minimumLeafSize,
+                             minimumGainSplit, maxDepth, IO::HasParam("warm_start"));
+        Timer::Stop("et_training");
+    }
+        if (IO::HasParam("test"))
+        {
+            mat testData = move(IO::GetParam<mat>("test"));
+            Timer::Start("rf_prediction");
+            // Get Predictions and probabilities.
+            arma::Row<size_t> predictions;
+            arma::mat probabilities;
+            etModel->et.Classify(testData, predictions, probabilities);
+        
+        // Save the outputs.
+        IO::GetParam<arma::mat>("probabilities") = std::move(probabilities);
+        IO::GetParam<arma::Row<size_t>>("predictions") = std::move(predictions);
+    }
+    // Save the output model.
+    IO::GetParam<ExtraTreesModel *>("output_model") = etModel;
+}
```

Comment:
The code in this file is very hard to read and follow because its style is inconsistent---if you can take a pass and improve it, I'd really appreciate it. :+1: